### ISSUE TITLE
Question on backward pass of the FusedDense layer

### ISSUE THREAD

**Issue by macrocredit** [Dao-AILab/flash-attention#426]
Hi - I was wondering when the backward pass of the FusedDense Layer will get called?

@staticmethod
@custom_bwd
def backward(ctx, grad_output, *args):
  ...
 
Because I didn't see that backward function above being get called when loss.backward() is called. Thanks.

**Comment by tridao** [comment]
Pytorch autograd will call that function if you use FusedDense forward in the forward pass. You can put in print statements / breakpoints to make sure that the forward and backward functions are being called.

**Comment by macrocredit** [comment]
OK; I think I figured it out. I think I need torch.is_autocast_enabled() == True for the backward pass to get called for some reasons. I am not sure why.
